draft-ietf-quic-multipath.md

```diff
@@ -1432,26 +1414,39 @@ MAX_PATHS frames are formatted as shown in {{fig-max-paths-frame-format}}.
 ~~~
 MAX_PATHS Frame {
   Type (i) = 0x15228c0b,
-  Maximum Paths (i),
+  Maximum Path Identifier (i),
 }
 ~~~
 {: #fig-max-paths-frame-format title="MAX_PATHS Frame Format"}
 
 MAX_PATHS frames contain the following field:
 
 Maximum Path Identifier:
-: A count of the cumulative number of path that can be opened
-  over the lifetime of the connection. This value cannot exceed 2^32-1, as it is not
-  possible to encode Path IDs larger than 2^32-1. Receipt of a frame that permits
-  opening of a path with Path ID larger than this limit MUST be treated
-  as a connection error of type FRAME_ENCODING_ERROR.
+: The latest Path ID that can be used for this connection.
+  This value cannot exceed 2^32-1, as it is not
+  possible to encode Path IDs larger than 2^32-1.
+
+Receipt of PATH_AVAILABLE, PATH_STANDBY, PATH_ABANDON or MP_ACK frames
+that uses a Path ID that is larger than the announced Path ID
+MUST be treated as a connection error of type FRAME_ENCODING_ERROR.
+
 
 Loss or reordering can cause an endpoint to receive a MAX_PATHS frame with
-a lower path limit than was previously received. MAX_PATHS frames that
-do not increase the path limit MUST be ignored.
+a smaller Path ID than was previously received. MAX_PATHS frames that
+do not announce a larger Path ID than previously received MUST be ignored.
+
+Endpoints SHOULD NOT issue new connection IDs which have path identifiers larger than
+the Path ID announced in the Maximum Path Identifier field in the
+MP_MAX_PATHS frame {{max-paths-frame}}.
+If no MAX_PATHS frame was received yet, the maximum Path ID
+correspondes to value of initial_max_paths transport parameter.
+
+An endpoint MUST NOT initiate a path with a Path ID larger than the Maximum Path Identifier value.
+An endpoint MUST terminate the a connection with an error of type MP_PROTOCOL_VIOLATION
+if a peer opens a path with a larger Path ID than permitted.
 
-An endpoint MUST NOT initiate a path with a Path ID higher than the Maximum Paths value.
-An endpoint MUST terminate the a connection with an error of type MP_PROTOCOL_VIOLATION if a peer opens more paths than was permitted.
+When there are not enough unused path identifiers, endpoints can
+send a MAX_PATHS frame to inform the peer that new path identifiers are available.
 
```

Comment:
For information, I struggled with the same set of issues when writing the "even numbers" PR. The text that I came up with was:
```
Maximum Paths:
: A count of the cumulative number of paths that can be opened
over the lifetime of the connection.

The endpoint receiving this frame MAY send MP_NEW_CONNECTION_ID frames
with Path IDs strictly lower than twice the Maximum Paths.
For example, if an endpoint received a Maximum Paths value of 3,
it may send MP_NEW_CONNECTION_ID with Path ID 0, 2 and 4, but not higher.

The Maximum Paths value cannot exceed 2^31-1, as it is not
possible to encode Path IDs larger than 2^32-2. Receipt of a
Path Identifier larger than this limit MUST be treated
as a connection error of type FRAME_ENCODING_ERROR.

The Maximum Paths value MUST NOT be lower than the value
advertised in the "initial_max_paths" transport parameter. Receipt
of a value lower than the "initial_max_paths" transport parameter
MUST be treated as a connection error of type MP_PROTOCOL_ERROR.
```
Some of that text was motivated by the odd-even split, but a lot of it was about clarify how the path-ID is used.

Reply:
We can define the max paths as either the cumulative number of path over the whole life or the largest id. The part that I was trying to fix is that the frame figure said the field is called "Maximum Paths" but the text was defining "Maximum Path Identifier". I thought the later is clearer but in the end we just have to fix it such that it is correct, one or the other way. @huitema do you prefer "Maximum Paths"?

Reply:
MAX_PATHS, MAX_PATH_ID, "bonnet blanc et blanc bonnet", as we say in French. I don't really care.

Reply:
Okay, but I guess if we use the max path id field, we should also rename the frame... 

Looking at this now again I guess renaming the field in the definition to "Maximum Paths" would have been the smaller chance. I'll revise this PR again tomorrow in order to keep chances as small as possible for now (but ensure things are correct). Sorry for forth and back.